lib/split/user.rb

```diff
@@ -38,28 +48,41 @@ def cleanup_old_versions!(experiment)
 
     def active_experiments
       experiment_pairs = {}
-      user.keys.each do |key|
-        Metric.possible_experiments(key_without_version(key)).each do |experiment|
+      keys_without_finished(user.keys).each do |key|
+        Metric.possible_experiments(Split::Experiment.key_without_version_for_experiment_key(key)).each do |experiment|
           if !experiment.has_winner?
-            experiment_pairs[key_without_version(key)] = user[key]
+            experiment_pairs[Split::Experiment.key_without_version_for_experiment_key(key)] = user[key]
           end
         end
       end
       experiment_pairs
     end
 
+    def key_for_experiment(experiment)
+      keys.find { |k| k.match(Regexp.new("^#{experiment.name}"))}
+    end
+
+    def version_for_experiment(experiment)
+      kfe = key_for_experiment(experiment)
+      if kfe
+        kfe.split(":").last
+      else
+        0
+      end
+    end
+
     private
 
+    def key_for_experiment_name(experiment_name)
```

Comment:
Ditto on concretely defining which key should be expected to come out of this function.

Reply:
same issue as above. This can be best thought of as belonging to the join table of the relation between user and experiment, were one to exist.